experiments/AtmosLES/schar_scalar_advection.jl

```diff
@@ -0,0 +1,287 @@
+#    year = {2002},
+#    month = {10},
+#    issn = {0027-0644},
+#    doi = {10.1175/1520-0493(2002)130<2459:ANTFVC>2.0.CO;2},
+#    url = {https://doi.org/10.1175/1520-0493(2002)130<2459:ANTFVC>2.0.CO;2},
+#}
+
+# ## [Initial Conditions]
+function init_schar!(bl, state, aux, (x, y, z), t)
+    ## Problem float-type
+    FT = eltype(state)
+
+    ## Unpack constant parameters
+    R_gas::FT = R_d(bl.param_set)
+    c_p::FT = cp_d(bl.param_set)
+    c_v::FT = cv_d(bl.param_set)
+    p0::FT = MSLP(bl.param_set)
+    _grav::FT = grav(bl.param_set)
+    γ::FT = c_p / c_v
+
+    c::FT = c_v / R_gas
+    c2::FT = R_gas / c_p
+
+    Tiso::FT = 250.0
+    θ0::FT = Tiso
+
+    ## Calculate the Brunt-Vaisaila frequency for an isothermal field
+    ## Hydrostatic background state
+    Brunt::FT = _grav / sqrt(c_p * Tiso)
+    Brunt2::FT = Brunt * Brunt
+    g2::FT = _grav * _grav
+
+    π_exner::FT = exp(-_grav * z / (c_p * Tiso))
+    θ::FT = θ0 * exp(Brunt2 * z / _grav)
+    ρ::FT = p0 / (R_gas * θ) * (π_exner)^c
+
+    ## Compute perturbed thermodynamic state:
+    T = θ * π_exner
+    e_int = internal_energy(bl.param_set, T)
+    ts = PhaseDry(bl.param_set, e_int, ρ)
+
+    ## Initial velocity
+    z₁::FT = 4000
+    z₂::FT = 5000
+    u₀::FT = 10
+    zscale = (z - z₁) / (z₂ - z₁)
+    if z₂ <= z
+        u = FT(1)
+    elseif z₁ <= z <= z₂
+        u = (sinpi(zscale / 2))^2
+    else
+        u = FT(0)
+    end
+    u *= u₀
+
+    ## Initial scalar anomaly profile
+    ## Equivalent to a nondiffusive tracer
+    Ax::FT = 25000
+    Az::FT = 3000
+    x₀::FT = 25000
+    z₀::FT = 9000
+    r = ((x - x₀) / Ax)^2 + ((z - z₀) / Az)^2
+    if r <= 1
+        χ = (cospi(r / 2))^2
+    else
+        χ = 0
+    end
+
+    ## State (prognostic) variable assignment
+    e_kin = FT(0)                                       # kinetic energy
+    e_pot = gravitational_potential(bl.orientation, aux)# potential energy
+    ρe_tot = ρ * total_energy(e_kin, e_pot, ts)         # total energy
+
+    state.ρ = ρ
+    state.ρu = SVector{3, FT}(ρ * u, 0, 0)
+    state.ρe = ρe_tot
+    state.tracers.ρχ = ρ * SVector{1, FT}(χ)
+end
+
+# Define a `setmax` method
+function setmax(f, xmax, ymax, zmax)
+    function setmaxima(xin, yin, zin)
+        return f(xin, yin, zin; xmax = xmax, ymax = ymax, zmax = zmax)
+    end
+    return setmaxima
+end
+
+function warp_schar(
+    xin,
+    yin,
+    zin;
+    xmax = 150000.0,
+    ymax = 5000.0,
+    zmax = 25000.0,
+)
+    FT = eltype(xin)
+    a::FT = 25000 ## Half-width parameter [m]
+    r = sqrt(xin^2 + yin^2)
+    h₀::FT = 3000 ## Peak height [m]
+    λ::FT = 8000 ## Wavelength
+    h_star =
+        abs(xin - xmax / 2) <= a ? h₀ * (cospi((xin - xmax / 2) / 2a))^2 : FT(0)
+    h = h_star * (cospi((xin - xmax / 2) / λ))^2
+    x, y, z = xin, yin, zin + h * (zmax - zin) / zmax
+    return x, y, z
+end
+
+function config_schar(FT, N, resolution, xmax, ymax, zmax)
+    u_relaxation = SVector(FT(20), FT(0), FT(0))
```

Comment:
Are you solving just the scalar equation or also the dynamics? If you are only solving the scalar equation with a given and constant velocity field, then you don't need the sponge because there are no gravity waves.

Reply:
Dynamics + scalar equation.